attrs.go

```diff
@@ -86,5 +100,27 @@ func fileStatFromInfo(fi os.FileInfo) (uint32, *FileStat) {
 	// os specific file stat decoding
 	fileStatFromInfoOs(fi, &flags, fileStat)
 
+	// The call above will include the sshFileXferAttrUIDGID in case
+	// the os.FileInfo can be casted to *syscall.Stat_t on unix.
+	// If implementations of RequestServers Handlers want to implement
+	// setting uids and gids while running on non-unix systems or
+	// without using *syscall.Stat_t underneath on unix, we offer
+	// an alternative way to support this through the FileInfoUidGid
+	// interface.
+	// If fi implements FileInfoUidGid, retrieve Uid, Gid from it.
+	if fiExt, ok := fi.(FileInfoUidGid); ok {
+		flags |= sshFileXferAttrUIDGID
+		fileStat.UID = fiExt.Uid()
+		fileStat.GID = fiExt.Gid()
+	}
```

Comment:
Consider, you might want to look at https://github.com/pkg/sftp/blob/master/ls_formatting.go#L55-L64 and how the `runLs` is going to work with this as well.

It is a known edge-case that some clients _are_ parsing that field for data, even though they were explicitly told not to.

Reply:
I didn't came across this `runLs` formatting yet, but of course it should be adjusted as well.